### DISCUSSION TITLE
Question about potential use case

### DISCUSSION THREAD

**Discussion by frank-h-w** [dylanljones/pyrekordbox#103] [Q&A] [answered]
Hi,

I'm looking this project and I was wondering if given the "My Tags" metadata seem to be accessible with this project and XML as well, would it be possible to transpose/copy the "MyTags" data from a track from a streaming service (Beatportlink, Tidal, etc...) to a local Mp3 file ?

Thanks
Frank

**Comment by dylanljones** [comment]
Hey Frank, 
If i understand your question correctly you are looking for a package to read/write the metadata of the MP3 files themselves?
`pyrekordbox` only handles the data of the database of Rekordbox, which stores all metadata in separate files.
If you want to update the metadata of your MP3 files **in** Rekordbox, then the package is the solution for you!

Otherwise, you should look into packages for handling audio metadata, ID3 tags specifically for MP3 files. A really extensive and versatile package for that is [mutagen](https://github.com/quodlibet/mutagen). It can handle all sorts of audio metadata, but it can get a little cumbersome. Other projects you can check out are [audio-metadata](https://github.com/thebigmunch/audio-metadata) or [eyeD3](https://github.com/nicfit/eyeD3).

**Reply by frank-h-w** [reply]
Thanks for the reply. So specifically my use case would be build my own script to copy existing "my tags" data stored on streaming tracks (tidal) that aren't stored in the XML database (I checked with an XML editor). As I slowly convert my library for standalone play via physical MP3 I'd like to be able to recuperate my Metadata stored in rekordbox for a track on tidal and copy that to the physical MP3 or AAC copy without having to do it manually via rekordbox interface. Lexicon the tool allows this for beatport link but not for Tidal. I figured I might be able to batch it if I had a way to retrieve the data. I found this project since rekordbox support confirmed to me the mytags data isn't in the XML but directly in the rekordbox database.

Is pyrekorbox still the way to go?

**Reply by dylanljones** [reply, marked as answer]
I am not sure what you are trying to achieve. I think you are confusing "My Tags" with the metadata of audio files. "My Tags" are user-defined attributes that you can add to tracks in the Rekordbox collection. These values have nothing to do with tracks on streaming platforms (like Tidal) since they are user-defined. 
However, you *can* access the "My tags" data with pyrekordbox:

````python
from pyrekordbox import Rekordbox6Database

db = Rekordbox6Database()
content = db.get_content()[10]  # Get a track from the database
for mytag_song in db.get_my_tag_songs(ContentID=content.ID):
    mytag = mytag_song.MyTag
    print(mytag.Name)
````

**Reply by frank-h-w** [reply]
thanks that's exactly what i was looking for!
